src/Translator.php

```diff
@@ -0,0 +1,542 @@
+            'silverstripe',
+            'cwp',
+            'symbiote',
+            'dnadesign'
+        ];
+        $vendorDir = dirname(dirname(dirname(__DIR__)));
+        foreach (scandir($vendorDir) as $vendor) {
+            if (!in_array($vendor, $vendorAllowList)) {
+                continue;
+            }
+            foreach (scandir("$vendorDir/$vendor") as $module) {
+                $modulePath = "$vendorDir/$vendor/$module";
+                if (!file_exists("$modulePath/.tx/config")) {
+                    continue;
+                }
+                $branch = $this->exec('git rev-parse --abbrev-ref HEAD', $modulePath);
+                if (!is_numeric($branch)) {
+                    throw new Exception("Branch $branch in $modulePath is not a minor or next-minor branch");
+                }
+                $this->modulePaths[] = $modulePath;
+            }
+        }
+    }
+
+    /**
+     * Backup local json files prior to replacing local copies with transifex
+     */
+    private function storeJson(): void
+    {
+        $this->log('Backing up local json files');
+        // Backup files prior to replacing local copies with transifex
+        $this->originalJson = [];
+        foreach ($this->modulePaths as $modulePath) {
+            $jsPath = $this->getJSLangDirectories($modulePath);
+            foreach ((array) $jsPath as $langDir) {
+                foreach (glob($langDir . '/src/*.json') as $path) {
+                    $str = file_get_contents($path);
+                    $this->originalJson[$path] = $this->jsonDecode($str);
+                }
+            }
+        }
+        $this->log('Finished backing up ' . count($this->originalJson) . ' json files');
+    }
+
+    /**
+     * Backup local yaml files in memory prior to replacing local copies with transifex
+     */
+    private function storeYaml(): void
+    {
+        $this->log('Backing up local yaml files');
+        $this->originalYaml = [];
+        foreach ($this->modulePaths as $modulePath) {
+            foreach (glob($this->getLangDirectory($modulePath) . '/*.yml') as $path) {
+                $rawYaml = file_get_contents($path);
+                $this->originalYaml[$path] = Yaml::parse($rawYaml);
+            }
+        }
+        $this->log('Finished backing up ' . count($this->originalYaml) . ' yaml files');
+    }
+
+    /**
+     * Set mtime to a year ago so that transifex will see these as obsolete
+     */
+    private function setJsonAndYmlFileTimes()
+    {
+        $date = date('YmdHi.s', strtotime('-1 year'));
+        foreach ($this->modulePaths as $modulePath) {
+            $name = $this->getModuleName($modulePath);
+            $this->log("Pulling sources from transifex for <info>{$name}</info> (min %{$this->txMinimumPerc} delta)");
+            $ymlLang = $this->getLangDirectory($modulePath);
+            if ($ymlLang) {
+                $this->exec("find $ymlLang -type f \( -name \"*.yml\" \) -exec touch -t $date {} \;");
```

Comment:
```suggestion
                $this->exec("find $ymlLang -type f \\( -name \"*.yml\" \\) -exec touch -t $date {} \\;");
```
This might not be explicitly necessary but it makes the intention a lot clearer that these _should_ be blackslashes in the executed command. Unless they shouldn't be, in which case what are they doing here?

Same with any other command strings like this.

Reply:
I works so I'm inclined to not change this fragile looking command